### PR TITLE
chore: update load gen to use new error handling

### PR DESCRIPTION
**Pull request by cprice404**
This gets rid of some straggling try/catch blocks in the loadgen and ensures that it will count throttles and other response types properly.

### CODE REVIEW

File: examples/MomentoLoadGen/Program.cs [hunk 1]
```diff
@@ -237,77 +236,87 @@ private async Task IssueAsyncSetGet(SimpleCacheClient client, CsharpLoadGenerato
...
-                switch (innerException!.StatusCode)
-                {
+                return Tuple.Create<AsyncSetGetResult, CacheGetResponse?>(AsyncSetGetResult.SUCCESS, miss);
+            } else if (getResponse is CacheGetResponse.Error error)
+            {
+                return Tuple.Create<AsyncSetGetResult, CacheGetResponse?>(ConvertErrorToAsyncSetGetResult(error.ErrorCode, error.Exception), null);
+            } else
+            {
+                throw new ApplicationException($"Unsupported get response: {getResponse}");
+            }
```
Comment (eaddingtonwhite) [inline examples/MomentoLoadGen/Program.cs:280]: total nit but I prefer switch statement over the multiline if. Not sure if c# reason though for doing in IF is cleaner with how we pass back error codes so not blocker at all. Just personal preference thing for me. Saw we had switch on status before.

File: examples/MomentoLoadGen/Program.cs [hunk 2]
```diff
@@ -360,13 +370,16 @@ internal class Program
...
-                       options.TimestampFormat = "hh:mm:ss ";
-                   }));
+            return LoggerFactory.Create(builder => {
+                builder.AddSimpleConsole(options =>
+                {
+                    options.IncludeScopes = true;
+                    options.SingleLine = true;
+                    options.TimestampFormat = "hh:mm:ss ";
+                });
+                builder.AddFilter("Grpc.Net.Client.Internal.GrpcCall", LogLevel.Error);
```
Comment (malandis) [inline examples/MomentoLoadGen/Program.cs:380]: This is really nice now

**Review by malandis — APPROVED** [review]

**Review by pgautier404 — APPROVED** [review]
LGTM!

File: examples/MomentoLoadGen/Program.cs (hunk 1 above)
Comment (cprice404) [reply examples/MomentoLoadGen/Program.cs:280]: Unfortunately the blessed/idiomatic way to do this kind of pattern matching in C# today is with if statements:

https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/pattern-matching#type-tests

I would rather have it as a switch too, but the smart casting to the local variable is kind of a crucial part of the user experience here now.

Comment (eaddingtonwhite) [reply examples/MomentoLoadGen/Program.cs:280]: ack figured was something like this.

**Pull request merged**